### ISSUE TITLE
Worker broken in 0.12.7 because Object.assign not available

### ISSUE THREAD

**Issue by ggoodman** [c9/c9.ide.language#8]
Use of `Object.assign` breaks any c9 instance running on node `0.12.x`.

Relevant line: https://github.com/c9/c9.ide.language/blob/master/worker.js#L1705

**Comment by nightwing** [comment]
worker.js shouldn't be loaded in the server, and i don't see this error when running c9 with node 0.12. 
What error do you see printed to console? Maybe full stacktrace can give a clue how this happens.

**Comment by ggoodman** [comment]
I'm seeing this issue running c9 local on OSX. Here is a screen capture.
![image](https://cloud.githubusercontent.com/assets/50030/12226651/52eb1976-b7ed-11e5-93bb-5b70c9af26c5.png)

``` bash
beauvallon% node --version
v0.12.9
beauvallon% node -e 'console.log(typeof Object.assign)'
undefined
```

**Comment by Pike** [comment]
Which browser are you using?

https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/Object/assign#Browser_compatibility says that we should have wide-spread support for Object.assign in browsers (and yeah, this should run in the browser, not in node)

**Comment by ggoodman** [comment]
This is being run in `nw.js` (local).

**Comment by nightwing** [comment]
@ggoodman we'll add a shim for Object.assign for old browsers, thanks for reporting

**Comment by ggoodman** [comment]
Fixed in c9/core@d77ae7459fc673e85ba53395690f806e9df33fae
